tests/testlibrary.c

```diff
@@ -1542,6 +1542,55 @@ test_list_updates (void)
   g_assert_true (res);
 }
 
+static void
+test_list_updates_offline (void)
+{
+  g_autoptr(FlatpakInstallation) inst = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GPtrArray) refs = NULL;
+  g_autoptr(FlatpakInstalledRef) runtime_ref = NULL;
+  g_autoptr(FlatpakRemote) remote = NULL;
+  gboolean res;
+  g_autofree char *saved_url = NULL;
+  g_autofree char *wrong_url = NULL;
+
+  inst = flatpak_installation_new_user (NULL, &error);
+  g_assert_no_error (error);
+
+  /* Install a runtime */
+  runtime_ref = flatpak_installation_install (inst,
+                                              repo_name,
+                                              FLATPAK_REF_KIND_RUNTIME,
+                                              "org.test.Platform",
+                                              NULL, NULL, NULL, NULL, NULL,
+                                              &error);
+  g_assert_no_error (error);
+  g_assert (FLATPAK_IS_INSTALLED_REF (runtime_ref));
+
+  /* Make the URL wrong to simulate being offline */
```

Comment:
I’m not sure that’s a good simulation of being offline, since all the internal error handing will be handling error 404s, rather than address resolution errors. Maybe just rename this test to refer to ‘empty resultset’ rather than ‘being offline’?

Reply:
Hmm, would be neat if we could do what `flatpak run --unshare=network` does to better simulate being offline.

Reply:
```c
if (g_test_subprocess ())
  {
    unshare (CLONE_NEWNET);
    /* rest of test */
  }

g_test_trap_subprocess (NULL, 0, 0);
g_test_trap_assert_passed ();
}
```

?

Reply:
Neat, didn't know about `g_test_trap_subprocess()`. I'll try that patch.